js/sky/src/grids/columnpicker.js

```diff
@@ -77,6 +83,44 @@
         $scope.locals.selectedColumns = initialSelectedColumns.slice(0);
         $scope.columnPickerHelpKey = columnPickerHelpKey;
         $scope.listMode = listMode;
+        if (colPickerSearchProperties) {
+            $scope.colPickerSearchProperties = colPickerSearchProperties;
+            $scope.searchCallback = function (args) {
+                var i,
+                    j, 
+                    n,
+                    m, 
+                    p,
+                    val,
+                    item, 
+                    items = $scope.initialColumns,
+                    filteredItems = [],
+                    searchTextUpper = angular.isString(args.searchText) ? args.searchText.toUpperCase() : '';
+                for (i = 0, n = items.length; i < n; i++) {
+                    item = items[i];
+                        
+                    if (!args.category || item.category === args.category) {
+                        if (!searchTextUpper) {
+                            filteredItems.push(item);
+                        } else {
+                            for (j = 0, m = $scope.colPickerSearchProperties.length; j < m; j++) {
+                                p = $scope.colPickerSearchProperties[j];
+                                if (item.hasOwnProperty(p)) {
+                                    val = item[p];
+
+                                    if (angular.isString(val) && val.toUpperCase().indexOf(searchTextUpper) >= 0) {
+                                        filteredItems.push(item);
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                
+                $scope.columns = translateColumnsToChecklistObjects(filteredItems); 
+            };
+        }
```

Comment:
I would have some concerns that this code would be a burden to maintain, since it mimics the checklist search code, right?  I wouldn't want to put in the need to maintain two different search functions.  Are my concerns valid? I feel like the search should be able to use some shared code between the two.